src/seedu/addressbook/AddressBook.java

```diff
@@ -202,13 +202,17 @@ public static void main(String[] args) {
         showWelcomeMessage();
         processProgramArgs(args);
         loadDataFromStorage();
-        while (true) {
+        renderInputTillExit();
```

Comment:
Yes here we can extract the method to keep the abstraction level single. Think about the other parts of your changes: do those affect the level of abstraction?